osu.Game/Beatmaps/WorkingBeatmap.cs

```diff
@@ -79,100 +78,97 @@ protected virtual Track GetVirtualTrack(double emptyLength = 0)
         /// <returns>The applicable <see cref="IBeatmapConverter"/>.</returns>
         protected virtual IBeatmapConverter CreateBeatmapConverter(IBeatmap beatmap, Ruleset ruleset) => ruleset.CreateBeatmapConverter(beatmap);
 
-        public virtual IBeatmap GetPlayableBeatmap(RulesetInfo ruleset, IReadOnlyList<Mod> mods = null, TimeSpan? timeout = null)
+        public virtual IBeatmap GetPlayableBeatmap(RulesetInfo ruleset, IReadOnlyList<Mod> mods = null, CancellationToken cancellationToken = default)
```

Comment:
Is it intentional that this will now never time out when the caller doesn't provide a cancellation token? i.e has all cases of calling this method been confirmed to not require timing out?

Reply:
You are welcome to confirm yourself.

Reply:
Well I mean this does change behaviour, since in previous code the load would be interrupted after 10 seconds by default in 99% of usages (as in every single call to `GetPlayableBeatmap()` in game other than the diffcalc path).

Reply:
Thanks for confirming. That's a weird hidden detail. I guess we do want that to return.

Reply:
I've added back the default along with testing. Not sure if it should be a nullable token or an overload method signature. Overload is a bit hard because `CancellationToken` is generally expected to be the last parameter, which means we can't also have an overload with an optional `TimeSpan` or `int timeout`.

Reply:
I think overload is fine. On the other side anything you could do via a `timeout` parameter, you can also do with a cancellation token. So I don't think that example is something to actually worry about.